crypto/x509/x509spki.c

```diff
@@ -131,3 +132,45 @@ char *NETSCAPE_SPKI_b64_encode(NETSCAPE_SPKI *spki) {
   OPENSSL_free(der_spki);
   return b64_str;
 }
+
+int NETSCAPE_SPKI_print(BIO *out, NETSCAPE_SPKI *spki) {
+  if (out == NULL || spki == NULL) {
+    OPENSSL_PUT_ERROR(X509, ERR_R_PASSED_NULL_PARAMETER);
+    return 0;
+  }
+  BIO_printf(out, "Netscape SPKI:\n");
+
+  // Print out public key algorithm and contents.
+  ASN1_OBJECT *spkioid;
+  X509_PUBKEY_get0_param(&spkioid, NULL, NULL, NULL, spki->spkac->pubkey);
```

Comment:
are we sure that `spki->spkac` is non-null?

Reply:
Added assertions for all the required non-null parameters at the top